### DISCUSSION TITLE
Register  file/Multi output RAM

### DISCUSSION THREAD

**Discussion by filipgaming008** [logisim-evolution/logisim-evolution#1375] [Ideas]
I havent noticed that there is a Register file in logisim yet, i know that i could make my own from flip flops or just using more RAM modules and connecting them to a single data-input and such, but this would take more of my computers resources and i dont know any java programming to make a library. It would be a nice feature to have that i guess.

**Comment by R3dst0ne** [comment]
There is a `Register` component in the `Memory` library.
![image](https://user-images.githubusercontent.com/62245695/150519070-e36e5c9f-3f9b-43b4-88c7-80ed6e146712.png)

Or are you looking for something else?

**Reply by filipgaming008** [reply]
Yeah i was looking for something else. Like for example you have registers and each register is connected to 1 Demux for the datain, and each register is connected to 2 or more Mux's for selecting the dataout for them. I should probably have explained it better in the first message. This is doable but im not sure if it will take much resources and i would like to have a lot more registers in that, like some sort of RAM. Each Mux can acces each register independently. Would be nice to have to connect to multiple components and not having to wait for a queue to acces data from just 1 data output.

**Reply by R3dst0ne** [reply]
What you describe sounds like a type of dual(multi)-port RAM.
There are are already [this](https://github.com/logisim-evolution/logisim-evolution/issues/79) issue and [this](https://github.com/logisim-evolution/logisim-evolution/discussions/838) discussion about the topic.

If you want to see something like this in LSe and have ideas how to implement it, you might want to revive the issue [79](https://github.com/logisim-evolution/logisim-evolution/issues/79).

**Reply by filipgaming008** [reply]
Oh, alright thanks.
